_posts/2024-08-08-cockpit-322.md

```diff
@@ -0,0 +1,104 @@
+---
+title: Cockpit 322
+author: mvo
+date: '2024-08-08'
+tags: cockpit
+slug: cockpit-322
+category: release
+summary: 'Host switcher is now deprecated and Cockpit Files adds permissions in details view'
+---
+
+Cockpit is the [modern Linux admin interface](https://cockpit-project.org/).
+We release regularly.
+
+Here are the release notes from Cockpit 322 and cockpit-files 5:
+
+## Shell: Deprecate host switching
+
+The "host switcher" at the top-left corner of Cockpit enables
+connections to multiple machines using SSH, runs a Cockpit session on
+each connected machine, and lets you switch between them.
+
+Although this feature is useful, when numerous hosts are active, they
+can potentially interact with each other in ways that might not be
+expected within an SSH trust boundary. For example, when connecting to
+an untrusted host from another host, it's assumed that the untrusted
+host should not have access or control over the originating
+host. However, due to web technology limitations, hosts may affect one
+another through [direct remote
+channels](https://issues.redhat.com/browse/COCKPIT-870), [iframe
+traversal](https://issues.redhat.com/browse/RHEL-4032), [cache
+poisoning](https://issues.redhat.com/browse/RHEL-4037), and other
+means.
+
+After several months of research into current web technologies, we
+found no way to prevent these concerns. Therefore, the Cockpit team
+intends to disable the host switcher feature.
+
+As a first step, the host switcher will be disabled by default in:
+- Fedora 41 and later
+- Debian Unstable and Testing
+- CentOS Stream 10
+- Red Hat Enterprise Linux 10
+- Arch Linux
+
+The host switcher will stay as-is in current stable Linux
+distributions. This includes Red Hat Enterprise Linux 9, CentOS Stream
+9, Fedora 39 and 40, Debian 12 "Bookworm", and Ubuntu 22.04 LTS and
+24.04 LTS.
+
+### Alternatives
+
+* The login page offers a "Connect to:" field which will directly
+  connect to another host using SSH. This is safe as long as you only
+  connect to a single host from a browser session. This will always be
+  supported. However, this only supports user/password or Kerberos
+  authentication, not SSH keys.
```

Comment:
```suggestion
* The login page offers a "Connect to:" field which will directly
  connect to another host using SSH. This is safe as long as you only
  connect to a single host within a browser session. This will always be
  supported. However, this only supports user/password or Kerberos
  authentication, not SSH keys.
```

Reply:
Thanks!

Reply:
Also: Logging out of a session fixes the issue, right? Is that implied here and clear enough?